src/types/interactions/application_command_option.ts

```diff
@@ -0,0 +1,22 @@
+import { SnakeCaseProps } from "../../util.ts";
+import { ApplicationCommandOptionChoice } from "./command_option_choice.ts";
+
+export interface ApplicationCommandOption {
+  /** Value of Application Command Option Type */
+  type: number;
```

Comment:
We should have a separate enum and cast that enum to the type of property "type".